KappaAnalysis/src/Producers/MuonCorrectionsProducer.cc

```diff
@@ -151,26 +133,30 @@ void MuonCorrectionsProducer::Produce(KappaEvent const& event, KappaProduct& pro
 			float scaledPy = muon->get()->p4.Py() * scaleFactor;
 			float scaledPz = muon->get()->p4.Pz() * scaleFactor;
 			float scaledE = TMath::Sqrt(TMath::Power(scaledPx,2) + TMath::Power(scaledPy,2) + TMath::Power(scaledPz,2) + TMath::Power(muonMass,2));
-
+			LOG(DEBUG) << "Corrected: " << "scaledPx: " << scaledPx << " scaledPy: " << scaledPy << " scaledPz: " 
+				<< scaledPz << " scaledE: " << scaledE << " Pt: " << muon->get()->p4.Pt() * scaleFactor;
 			muon->get()->p4.SetPxPyPzE(scaledPx, scaledPy, scaledPz, scaledE);
```

Comment:
Maybe move the debug output after the `p4.Set...` and get the values from the 4 vector, This way we make sure this function does the right thing as well.

Reply:
Done.